benchmark/QR_accuracy_stability_analysis.cc

```diff
@@ -15,148 +15,317 @@
 
 template <typename T>
 static void
-test_cond_helper_0(int64_t m, 
+error_check(int64_t m,
+            int64_t n,
+            int64_t k,
+            T norm_A,
+            T* A_dat,
+            T* A_1_dat,
+            T* Q_dat,
+            T* R_dat,
+            T* I_ref_dat
+            ) {
+
+    // Check orthogonality of Q
+    // Q' * Q  - I = 0
+    blas::gemm(Layout::ColMajor, Op::Trans, Op::NoTrans, k, k, m, 1.0, Q_dat, m, Q_dat, m, -1.0, I_ref_dat, k);
+    T norm_0 = lapack::lange(lapack::Norm::Fro, k, k, I_ref_dat, k);
+
+    // AP - QR
+    blas::gemm(Layout::ColMajor, Op::NoTrans, Op::NoTrans, m, n, k, 1.0, Q_dat, m, R_dat, k, -1.0, A_dat, m);
```

Comment:
This line of code makes me think that ``A_dat`` is really ``AP_dat``. If that's correct, please change the variable name and have your comments mention this.